components/script/dom/node.rs

```diff
@@ -2449,6 +2449,58 @@ impl VirtualMethods for Node {
         if let Some(list) = self.child_list.get() {
             list.as_children_list().children_changed(mutation);
         }
+
+        // Queue Mutation Observer records
+        match *mutation {
```

Comment:
I appreciate how nice it is to group these together here, but we really should be calling queue_a_mutation_record from places like https://dxr.mozilla.org/servo/rev/d1eef9d08a4dc3c7ff057797281d2b4f8d4f71c7/components/script/dom/node.rs#1619 instead. This will match https://dom.spec.whatwg.org/#concept-node-insert step 4 and step 7 better, for example.